recipes/slow-buffer-to-buffer-alloc-unsafe-slow/src/workflow.ts

```diff
@@ -0,0 +1,296 @@
+		const aliasMatches = text.matchAll(/SlowBuffer\s*:\s*(\w+)/g);
+		for (const match of aliasMatches) {
+			slowBufferAliases.add(match[1]);
+		}
+	}
+
+	// Find all named_imports that include SlowBuffer aliases (ESM imports)
+	const namedImports = rootNode.findAll({ rule: { kind: "named_imports" } });
+	for (const importNode of namedImports) {
+		const text = importNode.text();
+		// Match patterns like { SlowBuffer as SB } to extract alias "SB"
+		const aliasMatches = text.matchAll(/SlowBuffer\s+as\s+(\w+)/g);
+		for (const match of aliasMatches) {
+			slowBufferAliases.add(match[1]);
+		}
+	}
+
+	// Handle constructor patterns: new SlowBuffer(size) and new SB(size) for aliases
+	const constructorCalls = rootNode.findAll({
+		rule: {
+			kind: "new_expression",
+			has: {
+				field: "constructor",
+				kind: "identifier",
+				regex: "^SlowBuffer$",
+			},
+		},
+	});
+
+	for (const constructorCall of constructorCalls) {
+		const args = constructorCall.field("arguments");
+		if (args) {
+			// Extract the arguments text (without parentheses)
+			const argsText = args.text().slice(1, -1); // Remove ( and )
+			edits.push(constructorCall.replace(`Buffer.allocUnsafeSlow(${argsText})`));
+			hasChanges = true;
+		}
+	}
+
+	// Handle constructor calls with aliases
+	for (const alias of slowBufferAliases) {
+		const aliasConstructorCalls = rootNode.findAll({
+			rule: {
+				kind: "new_expression",
+				has: {
+					field: "constructor",
+					kind: "identifier",
+					regex: `^${alias}$`,
+				},
+			},
+		});
+
+		for (const constructorCall of aliasConstructorCalls) {
+			const args = constructorCall.field("arguments");
+			if (args) {
+				const argsText = args.text().slice(1, -1);
+				edits.push(constructorCall.replace(`${alias}.allocUnsafeSlow(${argsText})`));
+				hasChanges = true;
+			}
+		}
+	}
+
+	// Handle direct function calls: SlowBuffer(size)
+	const directCalls = rootNode.findAll({
+		rule: {
+			kind: "call_expression",
+			has: {
+				field: "function",
+				kind: "identifier",
+				regex: "^SlowBuffer$",
+			},
+		},
+	});
+
+	for (const directCall of directCalls) {
+		const args = directCall.field("arguments");
+		if (args) {
+			// Extract the arguments text (without parentheses)
+			const argsText = args.text().slice(1, -1); // Remove ( and )
+			edits.push(directCall.replace(`Buffer.allocUnsafeSlow(${argsText})`));
+			hasChanges = true;
+		}
+	}
+
+	// Handle direct function calls with aliases
+	for (const alias of slowBufferAliases) {
+		const aliasDirectCalls = rootNode.findAll({
+			rule: {
+				kind: "call_expression",
+				has: {
+					field: "function",
+					kind: "identifier",
+					regex: `^${alias}$`,
+				},
+			},
+		});
+
+		for (const directCall of aliasDirectCalls) {
+			const args = directCall.field("arguments");
+			if (args) {
+				const argsText = args.text().slice(1, -1);
+				edits.push(directCall.replace(`${alias}.allocUnsafeSlow(${argsText})`));
+				hasChanges = true;
+			}
+		}
+	}
```

Comment:
This scenario can be simplified using `utils/resolve-binding-path`, and improving findAll from ast-grep with [any](https://ast-grep.github.io/guide/rule-config/composite-rule.html#any) and [multiple matches](https://ast-grep.github.io/guide/api-usage/js-api.html#match).
```suggestion
  var ImportNode // importNode can be fetch with getNodeImportStatement or getNodeRequireCalls
  const binding = resolveBindPath(importN, "$.SlowBuffer") //it will resolve the name of the SlowBuffer in this file, independent of the way it was imported.
 	const aliasConstructorCalls = rootNode.findAll({
		rule: {
			any: [
				{
					kind: "new_expression",
					pattern: `new ${binding}($$$ARGS)`,
				},
				{
					kind: "call_expression",
					pattern: `${binding}($$$ARGS)`,
				},
			],
		},
	});

	for(const match of matches) {
		if(match.kind() === 'new_expression') { // each scenario can be handled by kind here if needed
			const args = match.getMultipleMatches() //get args
			edits.push(match.replace(/* handle replace */)
		}
	}

```